docs/source/1.0/spec/core/model.rst

```diff
@@ -2692,6 +2692,602 @@ after adding a member to the ``foo`` trait:
+===============
+
+The order of structure and union members is important for languages like C
+that require a stable ABI. Mixins provide a deterministic member ordering.
+Members inherited from mixins come before members defined directly in the
+shape.
+
+Members are ordered in a kind of depth-first, preorder traversal of mixins
+that are applied to a structure or union. To resolve the member order of a
+shape, iterate over each mixin applied to the shape in the order in which they
+are applied, from left to right. For each mixin, iterate over the mixins
+applied to the mixin in the order in which mixins are applied. When the
+evaluated shape has no mixins, the members of that shape are added to the
+resolved list of ordered members. After evaluating all the mixins of a shape,
+the members of the shape are added onto the resolved list of ordered members.
+This process continues until all mixins and the members of the starting shape
+are added to the ordered list.
+
+Given the following model:
+
+
+.. code-block:: smithy
+
+    @mixin
+    structure FilteredByName {
+        nameFilter: String
+    }
+
+    @mixin
+    structure PaginatedInput {
+        nextToken: String
+        pageSize: Integer
+    }
+
+    structure ListSomethingInput with [
+        PaginatedInput
+        FilteredByName
+    ] {
+        sizeFilter: Integer
+    }
+
+The members are ordered as follows:
+
+1. ``nextToken``
+2. ``pageSize``
+3. ``nameFilter``
+4. ``sizeFilter``
+
+
+Mixins on shapes with non-member properties
+===========================================
+
+Some shapes don't have members, but do have other properties. Adding a mixin
+to such a shape merges the properties of each mixin into the local shape. Only
+certain properties may be defined in the mixin shapes. See the sections below
+for which properties are permitted for each shape type.
+
+Scalar properties defined in the local shape are kept, and non-scalar
+properties are merged. When merging map properties, the values for local keys
+are kept. The ordering of merged lists / sets follows the same ordering as
+members.
+
+Service mixins
+--------------
+
+Service shapes with the :ref:`mixin-trait` may define any property. For
+example, in the following model:
+
+.. code-block:: smithy
+
+    operation OperationA {}
+
+    @mixin
+    service A {
+        version: "A"
+        operations: [OperationA]
+    }
+
+    operation OperationB {}
+
+    @mixin
+    service B with [A] {
+        version: "B"
+        rename: {
+            "smithy.example#OperationA": "OperA"
+            "smithy.example#OperationB": "OperB"
+        }
+        operations: [OperationB]
+    }
+
+    operation OperationC {}
+
+    service C with [B] {
+        version: "C"
+        rename: {
+            "smithy.example#OperationA": "OpA"
+            "smithy.example#OperationC": "OpC"
+        }
+        operations: [OperationC]
+    }
+
+The ``version`` property of the local shape is kept and the ``rename`` and
```

Comment:
Might simplify to say "The flattned equivalent of `C` with no mixins is:", and then remove the redefined operations since the other service shapes aren't defined either.

Reply:
> remove the redefined operations since the other service shapes aren't defined either

I'm not sure what you mean by this. The operations need to be there because they're still used.